dotCMS/src/main/java/com/dotcms/storage/model/Metadata.java

```diff
@@ -113,15 +126,33 @@ public boolean isImage(){
     }
 
     public int getWidth(){
-        return Try.of(()->(int)getFieldsMeta().get(WIDTH_META_KEY.key())).getOrElse(0);
+        final Object value = getFieldsMeta().get(WIDTH_META_KEY.key());
+        if(value instanceof Number){
+           final Number numericValue = (Number) value;
+           return numericValue.intValue();
+        }
+        Logger.debug(Metadata.class, ()->String.format("Invalid non numeric value found on metadata.width `%s` returning 0 ",value));
+        return 0;
     }
 
     public int getHeight(){
-        return Try.of(()->(int)getFieldsMeta().get(HEIGHT_META_KEY.key())).getOrElse(0);
+        final Object value = getFieldsMeta().get(HEIGHT_META_KEY.key());
+        if(value instanceof Number){
+            final Number numericValue = (Number) value;
+            return numericValue.intValue();
+        }
+        Logger.debug(Metadata.class, ()->String.format("Invalid non numeric value found on metadata.height `%s` returning 0 ",value));
+        return 0;
     }
 
     public long getModDate(){
-        return Try.of(()->(long)getFieldsMeta().get(MOD_DATE_META_KEY.key())).getOrElse(0L);
+        final Object value = getFieldsMeta().get(MOD_DATE_META_KEY.key());
+        if(value instanceof Number){
+            final Number numericValue = (Number) value;
+            return numericValue.longValue();
+        }
+        Logger.debug(Metadata.class, ()->String.format("Invalid non numeric value found on metadata.modDate `%s` returning 0 ",value));
+        return 0;
```

Comment:
we repeat this code several time, can we create a method with this logic and reuse it?